### PR TITLE
Rewrite main script for better style...

### PR DESCRIPTION
**Pull request by thinrope**
Now that python3 works, I spent some time to add a touch of style in the main bash script.

There is also a tiny change of functionality, so please have a look:
* directory names use proper ISO-8601 naming (instead of inventing yet-another-time-format)
* compressed data is stored in another directory, together
* by default the current directory is used for storage, not /tmp
* move all python into src/
* delete boilerplate restore_SOMETHING.sh scripts
* update README.md
* add some things to .gitignore

### CODE REVIEW

File: backup_grafana.sh [hunk 1]
```diff
@@ -1,34 +1,17 @@
...
+timestamp=$(date +"%FT%T")
+current_path=$(pwd)
+backup_dir="."
+archive_dir="./archive"
+
+for i in dashboards datasources folders
+do
+	F="${backup_dir}/${i}/${timestamp}"
+	[ -d "${F}" ] || mkdir -p "${F}"
+	P=$(echo ${i}| sed 's/^./save\U&/')
```
Comment (ysde) [inline backup_grafana.sh]: I tried this on my Macbook, this line create a `saveUdatasources.py`, which doesn't exist.

Can it only run on certain shell ?

Comment (thinrope) [reply backup_grafana.sh]: most probably a sed version thing... the `\U` syntax is not POSIX, only GNU :-/ Just to be sure, can you test this commend in your terminal:
````sh
echo dashboards | sed 's/^./save\U&/'
````

This sed hack (to cAmElCaSe `save{dashboards,datasources,folders}`) will not be needed, if we rename the files themselves to say `save_dashboards.py` and so on. Then I could replace the offending line with:
````sh
P="save_${i}"
````
or actually the loop would be simplified like:
````bash
for i in dashboards datasources folders
do
	F="${backup_dir}/${i}/${timestamp}"
	[ -d "${F}" ] || mkdir -p "${F}"
	python "${current_path}/save_${i}.py" "${F}"
done
````
There are probably 100 other ways (incl. hard-coding) to do it, not sure what is a portable one. If we are OK to rename the .py scripts, that will be better :+1: 
Let me know if you prefer a new/improved PR with all that in one (renaming the scripts, testing, etc.).

Comment (ysde) [reply backup_grafana.sh]: I'd like to make script as universal and can run on all environment as possible.
I don't know which naming convention is better in Python.
I think underscore `save_dashboards.py` is fine.

Please help for a improved PR :D
Thank you very much

**Comment by thinrope** [comment]
@ysde Please test at your site to confirm MacOS is fine :-D

**Pull request merged**